web/backend/utils/calc.ts

```diff
@@ -84,40 +84,42 @@ export const calculateMetrics = async ({ mappedMetrics, log }: FastifyInstance):
           if (metric.values.length > 0) {
             const [{ value, labels }] = metric.values
 
-            if (metric.name === 'process_resident_memory_bytes') {
-              aggregatedRss = calcBytesToMB(value)
+            if (isEntrypointService) {
+              if (metric.name === 'process_resident_memory_bytes') {
+                aggregatedRss = calcBytesToMB(value)
+              }
             }
 
             if (serviceId === labels.serviceId) {
               if (metric.name === 'nodejs_heap_size_total_bytes') {
-                serviceMemData.totalHeap = calcBytesToMB(value)
+                serviceMemData.totalHeap += calcBytesToMB(value)
                 aggregatedMemData.totalHeap += serviceMemData.totalHeap
               }
 
               if (metric.name === 'nodejs_heap_size_used_bytes') {
-                serviceMemData.usedHeap = calcBytesToMB(value)
+                serviceMemData.usedHeap += calcBytesToMB(value)
                 aggregatedMemData.usedHeap += serviceMemData.usedHeap
               }
 
               if (metric.name === 'nodejs_heap_space_size_used_bytes') {
                 metric.values.forEach(val => {
                   if (val.labels?.space === 'new') {
-                    serviceMemData.newSpace = calcBytesToMB(val.value)
+                    serviceMemData.newSpace += calcBytesToMB(val.value)
                     aggregatedMemData.newSpace += serviceMemData.newSpace
                   } else if (val.labels?.space === 'old') {
-                    serviceMemData.oldSpace = calcBytesToMB(val.value)
+                    serviceMemData.oldSpace += calcBytesToMB(val.value)
```

Comment:
summing memory is ok

Reply:
Nice! I updated the other parts for the "workers aggregated data". I can do the remaining mapping per single worker ID on a follow-up PR  😉

Reply:
@mcollina updated also the logic for single worker IDs